### PR TITLE
correct dependencies in a few Geometry packages

### PR DESCRIPTION
**Pull request by davidlange6**
missing reps noticed by modules build..

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

File: Geometry/HcalCommonData/BuildFile.xml [hunk 1]
```diff
@@ -1,6 +1,14 @@
 <use name="DetectorDescription/Core"/>
 <use name="DetectorDescription/DDCMS"/>
 <use name="DataFormats/HcalDetId"/>
+<use name="CondFormats/GeometryObjects"/>
```
Comment (ianna) [inline Geometry/HcalCommonData/BuildFile.xml:4]: @bsunanda - please, check if this dependency can be factored out. Thanks.

File: Geometry/HcalCommonData/BuildFile.xml [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
+<use name="DataFormats/DetId"/>
+<use name="DataFormats/Math"/>
+<use name="DetectorDescription/RegressionTest"/>
+<use name="FWCore/MessageLogger"/>
+<use name="FWCore/ParameterSet"/>
+<use name="FWCore/Utilities"/>
+<use name="SimDataFormats/CaloHit"/>
```
Comment (ianna) [inline Geometry/HcalCommonData/BuildFile.xml:11]: @bsunanda - please, check if this dependency can be factored out. Thanks.

**Review by ianna — COMMENTED** [review]
@davidlange6 - it looks like the new dependencies are slowly propagating to the packages. I think, we need to run `ignominy` to check the level of dependencies and restructure if needed.

**Comment by ianna** [comment]
@cvuosalo and @davidlange6 - I'd call it **de facto** dependencies, not sure if it's **de jure** :-)

**Comment by davidlange6** [comment]
They are actual #include dependencies...

On Sep 3, 2020, at 7:18 PM, Ianna Osborne <notifications@github.com<mailto:notifications@github.com>> wrote:



@cvuosalo<https://github.com/cvuosalo> and @davidlange6<https://github.com/davidlange6> - I'd call it de facto dependencies, not sure if it's de jure :-)

**Comment by cvuosalo** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**